packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.task-definition-enable-fault-injection.ts

```diff
@@ -0,0 +1,27 @@
+import * as cdk from 'aws-cdk-lib';
+import * as ecs from 'aws-cdk-lib/aws-ecs';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-ecs-task-definition-enable-fault-injection');
+
+const ec2TaskDefinition = new ecs.Ec2TaskDefinition(stack, 'Ec2TaskDefinition', {
+  enableFaultInjection: true,
+  networkMode: ecs.NetworkMode.HOST,
+});
+ec2TaskDefinition.addContainer('Ec2Container', {
+  image: ecs.ContainerImage.fromRegistry('public.ecr.aws/ecs-sample-image/amazon-ecs-sample:latest'),
+  memoryLimitMiB: 512,
+});
+
+const fargateTaskDefinition = new ecs.FargateTaskDefinition(stack, 'FargateTaskDefinition', {
+  enableFaultInjection: true,
+});
+fargateTaskDefinition.addContainer('FargateContainer', {
+  image: ecs.ContainerImage.fromRegistry('public.ecr.aws/ecs-sample-image/amazon-ecs-sample:latest'),
+  memoryLimitMiB: 512,
+});
+
+new IntegTest(app, 'TaskDefinitionEnableFaultInjection', {
```

Comment:
Could you add assertions to a `DescribeTaskDefinition` API call? You should be able to retrieve the `enableFaultInjection` prop there

Reply:
I'm struggling to add integration test assertions.

The enable fault injection was correctly set to true in the management console after deployment.

![image](https://github.com/user-attachments/assets/844e1a08-aee5-45c1-b611-8793934dd968)

However, the assertions are failing. 
I'm not yet sure of the root cause, but I might need to update the SDK version used in the integ tests.

```ts
test.assertions.awsApiCall('@aws-sdk/client-ecs', 'DescribeTaskDefinition', {
  taskDefinition: ec2TaskDefinition.taskDefinitionArn,
})
  .expect(ExpectedResult.objectLike({
    taskDefinition: {
      enableFaultInjection: true,
    },
  }));
```

```sh
  ASSERT     aws-ecs/test/integ.task-definition-enable-fault-injection-TaskDefinitionEnableFaultInjection/DefaultTest (undefined/us-east-1) 128.939s
      AssertionResultsAwsApiCallawssdkclientecsDescribeTaskDefinitionfd8b5aaee373c34a7ec063a10e6c89a0
{
  "tags": [],
  "taskDefinition": {
    "compatibilities": [ "EC2", "FARGATE" ],
    "containerDefinitions": [ { ... } ],
    "cpu": "256",
!!   Missing key 'enableFaultInjection'
    "enableFaultInjection": undefined,
```

Reply:
I don't think it's a SDK issue, but one with the API, or at least its docs, as [`DescribeTaskDefinition`](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTaskDefinition.html#API_DescribeTaskDefinition_ResponseSyntax) should return `enableFaultInjection`, whether or not it was set. I'm not getting the property from the CLI either with `aws ecs describe-task-definition`, with both the EC2 or the Fargate definition. 

Checking the task definition JSON in the web console does indicate that `enableFaultInjection` is being properly set by your integration, but I don't think we can test it as is. You could try to raise this issue with the AWS support but it might be better if a CDK team member could forward this issue.

Reply:
Thank you for the approval.
After updating the version in AWS CLI, I was able to retrieve it using describe-task-definition by AWS CLI.
Therefore, I think it might be an issue with a tool.

I'll investigate this further separately from this PR.​​​​​​​​​​​​​​​​

Reply:
You're absolutely right, I upgraded my CLI from `2.22.7` to `2.22.23` and the field now appears:

```sh
$ aws ecs describe-task-definition --task-definition <arn> | jq '.taskDefinition.enableFaultInjection'
true
```

I assumed the AWS API would do all the work with `describe*` commands, but I guess either the implementations map the results, or the API only returns certain fields after a certain version? Either way, kind of odd.

From what I can tell, since it's no longer documented, Node Lambdas ship with version `3.632.0` of the SDK (at least on `us-east-1`), so there's not much we can do about this as the assertions are ran on a custom resource. I don't think it's worth bundling shipping the SDK for the version lag we're experiencing, even if this version is over 4 months old